guac_core/src/storage/mod.rs

```diff
@@ -1,9 +1,15 @@
 use althea_types::EthAddress;
 use counterparty::Counterparty;
 use failure::Error;
+
 use futures;
+use futures::future::join_all;
 use futures::Future;
-use qutex::{FutureGuard, FutureWriteGuard, Guard, QrwLock, Qutex};
+
+use crypto::CryptoService;
+use CRYPTO;
+
+use qutex::{FutureGuard, Guard, QrwLock, Qutex};
```

Comment:
Did you verify how much overhead this dependency includes? What's special about a `qutex` and is it critical for your code to do its job? Will Rita release binary still oscillate around 8-10 megs?

Reply:
Well the problem which qutex solves is that it makes it a lot harder for futures based code to deadlock on mutexes and rwlocks, as other futures can make progress even when waiting to obtain a lock. I haven't been able to verify the overhead because guac doesn't actually build on mips at the moment, but I don't think it will be that significant as its only around 2kloc, and we have deps like ring or tokio which have orders of magnitude more